### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by ddriddle**
This pull request expands and harmonizes the README.rst and online documentation.

* Adds build check using twine to ensure the package and documentation will be accepted by PyPI.
* Defaults are now DRY.
* Removes unused command line option entity-id.

### CODE REVIEW

**Review by edthedev — APPROVED** [review]
Looks great. I really appreciate the documentation updates and clarifications.

**Review by mpitcel — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -73,27 +80,28 @@ one. For example::
...
-the IdP session expires::
+To switch roles, first log out, then log in again selecting a
+different role. Note that if you log in to the same IdP using the
+same username, you will not be prompted for your password or Duo
+factor until the IdP session expires::
 
     $ aws logout
     $ aws login
-    Username [netid]: 
+    Username [netid]:
```
Comment (kwessel) [inline README.rst:90]: @ddriddle, you state that the user won't be prompted again if their IdP session hasn't expired, but then your example shows them logging in again. Yes, this implies that their IdP session has expired, but it's slightly confusing. Suggest removing the username, password, and factor prompts from the logout and log back in example.

Comment (kwessel) [reply README.rst:90]: Never mind, I missed that there was only a username prompt, no prompts for password or second factor. This is fine.

**Review by kwessel — APPROVED** [review]

File: src/awscli_login/__init__.py [hunk 2]
```diff
@@ -41,34 +41,71 @@ class Login(BasicCommand):
...
         },
         {
-            'name': 'verbose',
-            'action': 'count',
-            'default': 0,
+            'name': 'duration',
+            'default': None,
             'cli_type_name': 'integer',
-            'help_text': 'Display verbose output'
+            'help_text': 'Time in seconds that the token will last'
```
Comment (JonRoma) [inline src/awscli_login/__init__.py:96]: Would it be clearer to say "Token lifetime in seconds"?

File: src/tests/util.py [hunk 3]
```diff
@@ -98,22 +98,27 @@ def error_handler(conn, func, *args, **kwargs):
...
+    password=None,
     factor=None,
-    force_refresh=False,
     passcode=None,
-    password=None,
-    refresh=0,
     role_arn=None,
-    subcommand=None,  # TODO FIXME!?
-    username=None,
+    # Advancded Properites (can NOT be set interactively)
```
Comment (JonRoma) [inline src/tests/util.py:110]: Typo: "Advanced Properties"

File: src/awscli_login/__init__.py [hunk 4]
```diff
@@ -41,34 +41,71 @@ class Login(BasicCommand):
...
         },
+        {
+            'name': 'http_header_factor',
+            'default': None,
+            'help_text': 'Name of HTTP Header to store the user\'s factor in'
```
Comment (JonRoma) [inline src/awscli_login/__init__.py:101]: This is nitpicky, but 'Name of HTTP header in which to store the user\'s factor' is cleaner.

File: src/awscli_login/__init__.py [hunk 5]
```diff
@@ -41,34 +41,71 @@ class Login(BasicCommand):
...
+        },
+        {
+            'name': 'http_header_passcode',
+            'default': None,
+            'help_text': 'Name of HTTP Header to store the user\'s passcode in'
```
Comment (JonRoma) [inline src/awscli_login/__init__.py:106]: This is nitpicky, but 'Name of HTTP header in which to store the user\'s password' is cleaner.

**Review by JonRoma — CHANGES_REQUESTED** [review]
@ddriddle, this looks good, and the changes I have suggested are very minor.

Longer term, I hope your team will be able to gradually add more comments inline with the code. This is a sophisticated and highly useful application, and I think description of the logic flow would be helpful for those trying to understand it.

File: README.rst [hunk 6]
```diff
@@ -149,8 +390,51 @@ To remedy the situation, change the data stored in the keyring as follows:
...
+location of the Windows install of awscli::
+
+    -bash: /mnt/c/Python39/Scripts/aws: c:\python39\python.exe^M: bad interpreter: No such file or directory
 
-Windows PowerShell, Command Prompt, and Git Shell for Windows are not
-currently supported because of problems with auto-renewal of AWS credentials,
-and other known issues.
+To remedy this issue please ensure that the location where the
+awscli is installed in the WSL comes before the location of the
+Windows install.
```
Comment (zcarrington) [inline README.rst:440]: Should we clarify here that WSL location needs to come before the Windows location in the PATH environment variable?
```suggestion
To remedy this issue please ensure that the location where the
awscli is installed in the WSL comes before the location of the
Windows install in the PATH environment variable.
```

**Review by zcarrington — APPROVED** [review]
Approved with a clarifying suggestion.

File: src/awscli_login/__init__.py (hunk 4 above)
Comment (ddriddle) [reply src/awscli_login/__init__.py:101]: If I did not want some nitpicking, I would not have asked you to review the documentation :-P Seriously, though thank you for the through proofreading!

**Review by zcarrington — APPROVED** [review]

**Review by mpitcel — APPROVED** [review]

**Pull request merged**